### PR TITLE
Deflake macos MixerFaultTest by broadening assertion ranges.

### PR DESCRIPTION
**Pull request by duderino**
**What this PR does / why we need it**:

Fix flake in macos tests that was introduced by https://github.com/istio/proxy/pull/2113

### CODE REVIEW

**Comment by kyessenov** [comment]
https://github.com/istio/proxy/issues/2124 is happening on linux. Does this help with that issue too?

File: test/integration/mixer_fault_test.cc [hunk 1]
```diff
@@ -545,16 +546,12 @@ TEST_F(MixerFaultTest, FailClosedAndSendPolicyResponseSlowly) {
   // Every response was a 5xx class
   EXPECT_EQ(0, client->class2xxResponses());
   EXPECT_EQ(0, client->class4xxResponses());
-  EXPECT_EQ(requests_to_send, client->class5xxResponses());
+  EXPECT_EQ(client->class5xxResponses(), requests_to_send);
```
Comment (JimmyCYJ) [inline test/integration/mixer_fault_test.cc]: I think EXPECT_EQ(requests_to_send, client->class5xxResponses()) is consistent with other EXPECT_EQ(expected result, tested result).

**Comment by JimmyCYJ** [comment]
> #2124 is happening on linux. Does this help with that issue too?

I think this addresses failures on macos.

File: test/integration/mixer_fault_test.cc [hunk 2]
```diff
@@ -563,6 +560,44 @@ TEST_F(MixerFaultTest, FailClosedAndSendPolicyResponseSlowly) {
   // Policy server closes every connection
   EXPECT_EQ(policy_cluster.connectionsAccepted(),
             policy_cluster.localCloses() + policy_cluster.remoteCloses());
+#else
+  // Macos is a bit flakier than Linux, so broaden assetion ranges to reduce
```
Comment (PiotrSikora) [inline test/integration/mixer_fault_test.cc]: Nit: it's `macOS`.

File: test/integration/mixer_fault_test.cc [hunk 3]
```diff
@@ -563,6 +560,44 @@ TEST_F(MixerFaultTest, FailClosedAndSendPolicyResponseSlowly) {
...
+  // test flakes.
+
+  // Most connections are successfully established.
+  EXPECT_IN_RANGE(client->connectSuccesses(), 0.8 * connections_to_initiate,
+                  connections_to_initiate);
+  EXPECT_IN_RANGE(client->connectFailures(), 0, 0.2 * connections_to_initiate);
```
Comment (PiotrSikora) [inline test/integration/mixer_fault_test.cc]: Could you add this as well?
```
EXPECT_EQ(client->connectSuccesses() + client->connectFailures(), connections_to_initiate);
```
Thanks!

File: test/integration/mixer_fault_test.cc (hunk 1 above)
Comment (PiotrSikora) [reply test/integration/mixer_fault_test.cc]: Yeah, I agree with @JimmyCYJ that the style in all those checks is a bit inconsistent.

But taking into account the range tests for macOS, I think that the tests above should move runtime-values to the left-hand side for readability, i.e.:
```
EXPECT_EQ(client->class2xxResponses(), 0);
EXPECT_EQ(client->class3xxResponses(), 0);
```

Comment (duderino) [reply test/integration/mixer_fault_test.cc]: The readability issues I will fix

(they are consistent with yoda conditions but in this case provide no actual value)

Comment (duderino) [reply test/integration/mixer_fault_test.cc]: done

File: test/integration/mixer_fault_test.cc (hunk 2 above)
Comment (duderino) [reply test/integration/mixer_fault_test.cc]: done

**Comment by duderino** [comment]
thanks @JimmyCYJ and @PiotrSikora

**Pull request merged**